src/containers/BooksForm.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import uniqid from 'uniqid'
+import { connect } from 'react-redux'
+import { createBook } from '../actions'
+
+const bookCategories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+]
+
+class BooksForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      title: '',
+      category: bookCategories[0],
+    }
+  }
+
+  handleChange = event => {
+    const { name, value } = event.target
+    this.setState({
+      [name]: value,
+    })
+  }
```

Comment:
does this update the state for `title` and `category` ??, looks like you only update `category`, CHECK AGAIN

compare with this
```
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}

```

Reply:
Hi! Thanks for the review 😄 . I think this is updating each one whenever each one changes.

The flow is the following:

1-. The `handleChange` method is called both by the `input` (the title) and the `select` (the category) when they change. Both of those have a name corresponding to the state (`<select name="title" ...)`. When either one of those changes, the `handleChange` method is triggered.

2-. On the `handleChange` method, we get the name and the value of the corresponding element which triggered the event: `const { name, value } = event.target`. Is the event was triggered by the `input`, then the `name` and `value` would be `name: 'title'` and `value: whatever the user typed in the field`. If was triggered by the `select`, then it is `name: 'select'`, `value: the selected item from the list`.

3-. We update the property by the name, so if the `select` changed, we update `state['category']`, otherwise, in the case of the `title`, we update `state['title']`, we only update the one that changed. No need to update the other if it hasn't changed.

That's the logic behind it, I hope this makes sense. I got the idea from the docs 😄 : https://reactjs.org/docs/forms.html#handling-multiple-inputs

Reply:
thanks for the ressource 😊